File: backend/album-collection/Models/Album.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace album_collection.Models
{
    public class Album
    {
        public int Id { get; set; }



        public string Name { get; set; }
        public int ArtistId { get; set; }

        public virtual Artist Artist { get; set; }

        public string Image { get; set; }

        // public string RecordLabel { get; set; }

        public virtual IEnumerable<Song> Songs { get; set; }


        public Album()
        {
        }

        public Album(int id, string name, int artistId, string image) // , string recordLabel)
        {
            Id = id;
            Name = name;
            ArtistId = artistId;
            Image = image;
            // RecordLabel = recordLabel;
        }
    }

       
}

File: frontend/src/js/components/Album.js
export default function Album(album){

    if(album.songs == null){
        album.songs = [];
    }
    
    return `
    <h1>${album.name}</h1>
    <input type = "hidden" id = "changeAlbumId" value = "${album.id}"/>
    <input type = "hidden" id = "changeAlbumArtistId" value = "${album.artistId}"/>
    <input type = "hidden" id = "changeAlbumImage" value = "${album.image}"/>
    <h2 class = "ArtistName" class = "artist_name" id = "${album.artist.id}">${album.artist.name}</h2>
    <img src=${album.image} alt=${album.name}>
    <br>
    <button class = "btnEditAlbum">Update Album</button>
        <ol>
        ${album.songs.map(song =>{
            return `
                <li>
                    <h2 class = "song_title" id = "${song.id}">${song.title}</h2>
                    <button class = "btnDeleteSong" id = "${song.id}">Delete Song</button>
                    </li>
                    <br>
                    <br>
                    <br>
                    <br>
                    <br>
                    <br>
                    <br>
                    <br>
            `
        }).join('')}
        </ol>
        <section class = "addSong">
            <label>Song Title: </label>
            <input type = "text" id = "newSongTitle" placeholder = "Song Title"/>
            <br>
            <label>Song Length: </label>
            <input type = "number" min ="1" id = "newSongLength" placeholder = "Song Length in Seconds"/>
            <br>
            <label>Link to Song: </label>
            <input type = "text" id = "newSongLink" placeholder = "Youtube URL"/>
            <br>
            <button class = "songAddButton" id = "${album.id}" >Add Song</button>
        </section>
        `




       }

